UI/src/model/selectedMassTransitStop.js

```diff
@@ -431,14 +499,24 @@
     var deleteMassTransitStop = function (poistaSelected) {
       if (poistaSelected) {
         var currAsset = this.getCurrentAsset();
-        backend.deleteAllMassTransitStopData(currAsset.id, function () {
-          eventbus.trigger('massTransitStopDeleted', currAsset);
-        }, function (errorObject) {
-          cancel();
-          if (errorObject.status == FAILED_DEPENDENCY_424) {
-            eventbus.trigger('asset:deleteTierekisteriFailed');
-          }
-        });
+        if(isServiceStop(currAsset.payload.properties)){
+          backend.deleteAllMassServiceStopData(currAsset.id, function () {
+            assetHasBeenModified = false;
+            eventbus.trigger('massTransitStopDeleted', currAsset);
+          }, function () {
+            cancel();
+          });
+        }else{
```

Comment:
Duplicated code again only because of different backend functions to call.

Reply:
Agree.
Check my suggestion code.

```javascript
        var payloadProperties = currAsset.payload.properties;
        var backendDeleteOperation = isServiceStop(payloadProperties) ? backend.deleteAllMassServiceStopData : backend.deleteAllMassTransitStopData;

        backendDeleteOperation(currAsset.id, function () {
          assetHasBeenModified = false;
          eventbus.trigger('massTransitStopDeleted', currAsset);
        }, function (errorObject) {
          cancel();
          if (!isServiceStop(payloadProperties) && errorObject.status == FAILED_DEPENDENCY_424) {
            eventbus.trigger('asset:deleteTierekisteriFailed');
          }
        })
```

Reply:
Done